lib/auth/export.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"encoding/pem"
+	"strings"
+	"time"
+
+	apidefaults "github.com/gravitational/teleport/api/defaults"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/sshutils"
+	"github.com/gravitational/trace"
+)
+
+// ExportAuthoritiesRequest has the required fields to create an export authorities request.
+// An empty AuthType exports all types.
+type ExportAuthoritiesRequest struct {
+	AuthType                   string
+	Client                     ClientI
```

Comment:
I find it a bit odd for the entire ClientI to be contained within the request. Maybe pass that in as a separate param?  (Assuming we don't want a type to handle this.)